### ISSUE TITLE
uploading sounds to catbox now produces bad filenames

### ISSUE THREAD

**Issue by vt-idiot** [rcc11/4chan-sounds-player#6]
# TL;DR
Uploading a file to catbox now produces a malformed filename that looks like this:
>`image[sound=%0A%0Ahttps%3A%2F%2Ffiles.catbox.moe%2Fsound.ext].ext`
## The Issue
I think Catbox's API response now looks like this? Or something like this:

>` `
>` `
>`https://files.catbox.moe/sound.ext`

## Things I tried

Various RegEx in both "Path" and "Match" fields. All result in uploads to "undefined" even though I can see that they are in fact uploading to catbox - my userhash is in the config, and the very first "undefined" failure did actually upload the file.
 
 `^\s|^[^-]*\b` `/^\s|^[^-]*\b/gm` `^\s|^[^-]*\/` and I don't remember what else. I tried adding `https://files.catbox.moe/%s` in the "File URL Format" box since I wasn't even sure what those RegEx would select, but that just gave me `https://files.catbox.moe/undefined` as output.

However,
>I am an idiot, please undastand

## The Fix?

I know it's as simple as editing the Sounds Player's default "Response Path" and "Response Match" fields for catbox in the "Hosts" configuration, and maybe "File URL Format", but once again:
>I am an idiot, please undastand

**Comment by rcc11** [comment]
Looks like there's a couple of line breaks before the url in the response. For now set "Response Match" to `.*https://(.*)`.  That'll match the URL (sans protocol) so "File URL Format" can be left blank to leave it as is. I'll have a more considered fix with the next release.

For the record `responseMatch` below is how the regex is used. So no slashes, and the first capture group is what's pulled out as the URL. If you set "File Url Format" (`host.soundUrl` below) that capture is put in place of %s in the provided format. This all could definitely be documented better.
https://github.com/rcc11/4chan-sounds-player/blob/957b2812d133347f2851098e2d1757bede3817e0/src/components/tools/create.js#L285-L290

**Comment by vt-idiot** [comment]
>  For now set "Response Match" to `.*https://(.*)`.

Thank you for responding so quickly, that worked like a charm!
Remember,

> I am an idiot, please undastand

The only part of how it works that I actually understand is the "File URL Format" field and how to use the %s variable, e.g. now I could stick `https://%s` in there if I still wanted https:// in my file names (I don't).
